core/src/main/scala/io/snappydata/util/ServiceUtils.scala

```diff
@@ -177,4 +177,34 @@ object ServiceUtils {
         }
     }
   }
+
+  /**
+   * We capture table ddl string and add it to table properties before adding to catalog.
+   * This will also contain passwords in the string such as jdbc connection string or
+   * s3 location etc. This method masks passwords
+   * @param str DDL string that might contain jdbc/s3 passwords
+   * @return similar string with masked passwords
+   */
+  def maskPasswordsInString(str: String): String = {
+    val jdbcPattern1 = ".*jdbc.*(?i)\\bpassword\\b=([^);&]*).*".r
```

Comment:
Factor out list of patterns at common place as constant.
See if we can add more or these patterns are sufficient. e.g other blob storage.